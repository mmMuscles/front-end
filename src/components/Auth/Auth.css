.footie {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    box-sizing: border-box;
    position: fixed;
    bottom: 30px;
    /* width: 100vw; */
    /* height: 150px; */
    color: beige;
    padding: 15px;
}
.footie > button {
    max-width: 10px;
}