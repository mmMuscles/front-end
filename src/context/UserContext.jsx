import { createContext, useContext, useMemo, useState } from 'react';
import { supabase } from '../services/supabaseClient';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const session = supabase.auth.session()
  const [profile, setProfile] = useState({})
  const [user, setUser] = useState(
    session ? { id: session.user.id, email: session.user.email } : {}
  );
  
  const value = useMemo(() => ({ user, setUser, profile, setProfile}), [user, profile]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }
  return context;
};

export { UserProvider, useUser };
