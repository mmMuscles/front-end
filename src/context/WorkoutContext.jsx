import { useContext, createContext, useState } from "react";



const workoutCtx = createContext();


export const WorkoutProvider = ({ children }) => {
    const [workouts, setWorkouts] = useState([])

return <workoutCtx.Provider value={{workouts, setWorkouts}}>{children}</workoutCtx.Provider>
}



export const useWorkout = () => {
    const context = useContext(workoutCtx)

    if(context === undefined){
        throw new Error('Invalid Provider')
    }

    return context;
}
