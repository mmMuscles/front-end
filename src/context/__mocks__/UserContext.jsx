import { createContext, useContext, useMemo, useState } from "react";

const UserContext = createContext();

const UserProvider = ({ children, mockUser }) => {
  const [profile, setProfile] = useState({});
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  const value = useMemo(
    () => ({ user, setUser, profile, setProfile }),
    [user, profile]
  );

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error("useUser must be used within a UserProvider");
  }
  return context;
};

export { UserProvider, useUser };
