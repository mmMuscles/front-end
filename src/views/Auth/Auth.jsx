import { useState } from 'react'
import { supabase } from '../../services/supabaseClient'


export default function Auth() {
  const [loading, setLoading] = useState(false)
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')

  const handleLogin = async (email, password) => {
    try {
      setLoading(true)
      const { error } = await supabase.auth.signUp({ email , password})
      if (error) throw error
      alert('Check your email for the login link!')
    } catch (error) {
      alert(error.error_description || error.message)
    } finally {
      setLoading(false)
    }
  }

  return (
    <div className="row flex flex-center">
      <div className="col-6 form-widget">
        <h1 className="header">Supabase + React</h1>
        <p className="description">Sign in via magic link with your email below</p>
        <div>
          <input
            className="inputField"
            type="email"
            placeholder="Your email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
           <input
            className="inputField"
            type="password"
            placeholder="Your password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
        <div>
          <button
            onClick={(e) => {
              e.preventDefault()
              handleLogin(email,password)
            }}
            className={'button block'}
            disabled={loading}
          >
            {loading ? <span>Loading</span> : <span>Send magic link</span>}
          </button>
        </div>
      </div>
    </div>
  )
}
