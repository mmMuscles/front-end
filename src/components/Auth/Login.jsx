import React, { useState } from 'react'
import { useHistory } from 'react-router-dom';
import { useUser } from '../../context/UserContext';
import { signInUser } from '../../services/supabaseClient';

export default function Login() {
    const [userEmail, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const history  = useHistory();
    const [isLoading, setLoading ] = useState(false);
    const { setUser } = useUser({id:'', email:''});

    const handleSignIn = async (e) => {
        e.preventDefault();
            try {
                setLoading(true);
                const auth = await signInUser(userEmail, password);
                await setUser({id:auth.id, email: auth.email})
                setLoading(false);
                const redirecting = '/profile' ;
                history.replace(redirecting)
            } catch (error) {
              throw error;
            }
          };
    
    return (
        <>
          {isLoading ? <h1>....Loading</h1> : null}
        <form className='flex px-6 mx-6'>
            <fieldset>
                {/* <legend>Sign-in</legend> */}
          <label htmlFor='email'>email </label>
          <input
            className="text-black rounded-sm opacity-70 my-1 mr-6 pl-1"
            type="email"
            id='email'
            required
            placeholder="Your email"
            value={userEmail}
            onChange={(e) => setEmail(e.target.value)}
          />
          <label htmlFor='password'>password </label>
           <input
            className="text-black rounded-sm opacity-70 my-1 pl-1"
            type="password"
            id='password'
            required
            placeholder="Your password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
          <button className='bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-4 mx-6 rounded'
          type='submit' 
          onClick={handleSignIn}>SignIn!</button>
          </fieldset>
        </form>
        </>
    )
}
