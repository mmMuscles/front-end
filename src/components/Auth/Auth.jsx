import { useState } from 'react';
import { Link } from 'react-router-dom';
import { useUser } from '../../context/UserContext';
import { logOutUser } from '../../services/supabaseClient';
import Login from '../Login';
import SignUp from '../SignUp';
import './Auth.css'


export default function Auth() {
  const { user, setUser } = useUser();
  const [ newUser, setNewUser ] = useState(true)

 const handleStatus = () => {
    setNewUser(!newUser)
 }
 

 const handleLogOut = async () => {
    setUser({ email:'', id:''})
    await logOutUser();
}
const authForm = (
    newUser
        ? <div className='flex'><Login className='form' /> <button className='bg-slate-700 opacity-90 hover:bg-blue-700 text-white font-bold py-1 px-4 rounded m-15' onClick={handleStatus}>Need to Sign Up?</button>
        </div>
        : <div className='flex'><SignUp className='form' />  <button className='bg-slate-700 opacity-90 hover:bg-blue-700 text-white font-bold py-1 px-4 rounded m-15' onClick={handleStatus}>Need to Login?</button>
        </div>
)

  return (
   <>
   <div className='footie'>
    {user.email ? <main>
              <h1>Hello, {user.email}</h1>
                <button className='bg-blue-300 hover:bg-blue-500 text-white font-bold py-2 px-4 rounded w-22 border-4'
                onClick={handleLogOut} >Log out</button>
                <Link to='/calendar' className='bg-slate-500 hover:bg-slate-700 rounded px-5 m-5' >Calendar</Link>
                <Link to='/profile' className='bg-slate-500 hover:bg-slate-700 rounded px-5 m-5' >Profile</Link>
          </main>: authForm}
   </div>
   </>
  )
}
