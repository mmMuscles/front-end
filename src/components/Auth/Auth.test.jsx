import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { UserProvider } from '../../context/UserContext';
import Auth from './Auth';

describe('Auth', () => {
    it('renders the auth page', () => {
        const { container } = render(
            <UserProvider>
                <MemoryRouter>
                    <Auth />
                </MemoryRouter>
            </UserProvider>
        );

        expect(container).toMatchSnapshot();
    })
})