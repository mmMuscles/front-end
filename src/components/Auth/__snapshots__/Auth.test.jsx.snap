// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test renders the top nav header 1`] = `
<div>
  <div
    class="footie"
  >
    <div
      class="flex"
    >
      <form
        class="flex px-6 mx-6"
      >
        <fieldset>
          <label
            for="email"
          >
            email 
          </label>
          <input
            class="text-black rounded-sm opacity-70 my-1 mr-6 pl-1"
            id="email"
            placeholder="Your email"
            required=""
            type="email"
            value=""
          />
          <label
            for="password"
          >
            password 
          </label>
          <input
            class="text-black rounded-sm opacity-70 my-1 pl-1"
            id="password"
            placeholder="Your password"
            required=""
            type="password"
            value=""
          />
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-4 mx-6 rounded"
            type="submit"
          >
            SignIn!
          </button>
        </fieldset>
      </form>
       
      <button
        class="bg-slate-700 opacity-90 hover:bg-blue-700 text-white font-bold py-1 px-4 rounded m-15"
      >
        Need to Sign Up?
      </button>
    </div>
  </div>
</div>
`;
